src/lib.rs

```diff
@@ -268,83 +269,84 @@ pub fn fbeta_score(pred: &[u64], actual: &[u64], beta: f32, average: Option<Stri
 /// ```
 /// use parsnip::jaccard_similiarity_score;
 /// 
-/// let actual = vec![0, 2, 1, 3];
-/// let pred = vec![0, 1, 2, 3];
+/// let actual = vec![0_u8, 2, 1, 3];
+/// let pred = vec![0_u8, 1, 2, 3];
 /// 
 /// assert_eq!(jaccard_similiarity_score(&pred, &actual), 0.5);
 /// ```
-pub fn jaccard_similiarity_score(pred: &[u64], actual: &[u64]) -> f32 {
-    categorical_accuracy(pred, actual)
+pub fn jaccard_similiarity_score<T: Unsigned + ToPrimitive>(pred: &[T], actual: &[T]) -> f32 {
+    return categorical_accuracy(pred, actual);
 }
 
 
+
 #[cfg(test)]
 mod tests {
     use super::*;
     #[test]
     fn test_gini() {
-        let vec = vec![0, 0, 0, 1];
+        let vec : Vec<usize> = vec![0, 0, 0, 1];
         assert_eq!(0.375, gini(&vec));
-        let v2 = vec![0, 0];
+        let v2 : Vec<usize> = vec![0, 0];
         assert_eq!(0.0, gini(&v2));
-        let mut v3 = vec![0];
+        let mut v3: Vec<usize> = vec![0];
         v3.pop();
         assert_eq!(1.0, gini(&v3));
```

Comment:
Unlike in Java, there is no convention to employ the left-handed member of an assertion as the "expected" value (and then the "obtained" value). There's no need to change what's already done, but don't feel inclined to follow this particular order.